operator/controllers/btpoperator_controller.go

```diff
@@ -89,27 +154,51 @@ func (r *BtpOperatorReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 		return ctrl.Result{}, r.Update(ctx, cr)
 	}
 
+	if !cr.ObjectMeta.DeletionTimestamp.IsZero() && cr.Status.State != types.StateDeleting {
+		return ctrl.Result{}, r.SetStatus(types.StateDeleting, ctx, cr)
+	}
+
 	switch cr.Status.State {
 	case "":
 		return ctrl.Result{}, r.HandleInitialState(ctx, cr)
 	case types.StateProcessing:
 		return ctrl.Result{}, r.HandleProcessingState(ctx, cr)
 	case types.StateError:
 		return ctrl.Result{}, r.HandleErrorState(ctx, cr)
+	case types.StateDeleting:
+		return ctrl.Result{}, r.HandleDeletingState(ctx, cr)
 	}
 
 	return ctrl.Result{}, nil
 }
 
-func (r *BtpOperatorReconciler) HandleInitialState(ctx context.Context, cr *v1alpha1.BtpOperator) error {
-	logger := log.FromContext(ctx)
-	logger.Info("Handling Initial state")
+func (r *BtpOperatorReconciler) reconcileRequestForAllBtpOperators(secret client.Object) []reconcile.Request {
+	btpOperators := &v1alpha1.BtpOperatorList{}
+	err := r.List(context.Background(), btpOperators)
+	if err != nil {
+		return []reconcile.Request{}
+	}
+	requests := make([]reconcile.Request, len(btpOperators.Items))
+	for i, item := range btpOperators.Items {
+		requests[i] = reconcile.Request{NamespacedName: k8sgenerictypes.NamespacedName{Name: item.GetName(), Namespace: item.GetNamespace()}}
+	}
+
+	return requests
+}
 
+func (r *BtpOperatorReconciler) SetStatus(new types.State, ctx context.Context, cr *v1alpha1.BtpOperator) error {
```

Comment:
We can shorten this method to two lines of code:
cr.SetStatus(status.WithState(newState))
return r.Status().Update(ctx, cr)
and call it UpdateBtpOperatorStatus()